src/lib/components/workflow/workflow-counts.svelte

```diff
@@ -1,90 +1,44 @@
 <script lang="ts">
   import { page } from '$app/stores';
 
-  import type { WorkflowFilter } from '$lib/models/workflow-filters';
-  import { workflowStatuses } from '$lib/models/workflow-status';
+  import { fetchWorkflowCountByExecutionStatus } from '$lib/services/workflow-counts';
   import { workflowFilters } from '$lib/stores/filters';
-  import { isStatusFilter } from '$lib/utilities/query/filter-search';
-  import { toListWorkflowQueryFromFilters } from '$lib/utilities/query/filter-workflow-query';
-  import { combineFilters } from '$lib/utilities/query/to-list-workflow-filters';
-  import { updateQueryParameters } from '$lib/utilities/update-query-parameters';
+  import { workflowCount, workflowsQuery } from '$lib/stores/workflows';
+  import type { WorkflowStatus } from '$lib/types/workflows';
+  import { decodePayload } from '$lib/utilities/decode-payload';
 
-  import WorkflowCount from './workflow-count.svelte';
+  import WorkflowCountAll from './workflow-count-all.svelte';
+  import WorkflowCountStatus from './workflow-count-status.svelte';
 
-  $: statusFilters = $workflowFilters.filter((filter) =>
-    isStatusFilter(filter.attribute),
-  );
+  $: namespace = $page.params.namespace;
+  let statusGroups: { status: WorkflowStatus; count: number }[] = [];
 
-  function mapStatusToFilter(value: string) {
-    return {
-      attribute: 'ExecutionStatus',
-      value,
-      operator: '',
-      parenthesis: '',
-      conditional: '=',
-    };
-  }
-
-  function mapStatusesToFilters(filters: WorkflowFilter[]) {
-    if (filters.length === 1) {
-      return [mapStatusToFilter(filters[0].value)];
-    } else {
-      return filters.map((filter, i) => {
-        const operator = i === filters.length - 1 ? '' : 'OR';
+  const fetchCounts = async () => {
+    const { count, groups } = await fetchWorkflowCountByExecutionStatus({
+      namespace,
+      filters: $workflowFilters,
+    });
+    $workflowCount.totalCount = parseInt(count);
+    statusGroups = groups
+      .map((group) => {
+        const status = decodePayload(
+          group?.groupValues[0],
+        ) as unknown as WorkflowStatus;
+        const count = parseInt(group.count);
```

Comment:
nit: `fetchWorkflowCountByExecutionStatus` could probably just do this data massaging

Reply:
I kinda like it in the view layer because we are massaging this data specifically for this component. Down the road we might want the service to do something else, so rather leave it more generic.

Reply:
Fair point 👍